tests/python_tests/test_static_llm_pipeline.py

```diff
@@ -0,0 +1,157 @@
+    pipe = ov_genai.LLMPipeline(model_path, "NPU", pipeline_config)
+
+
+pipeline_configs = [
+    { "MAX_PROMPT_LEN":   -1  },
+    { "MAX_PROMPT_LEN":   "1" },
+    { "MIN_RESPONSE_LEN": -1  },
+    { "MIN_RESPONSE_LEN": "1" }
+]
+@pytest.mark.skipif(is_npu_available, reason="Not supposed to be run on NPU device")
+@pytest.mark.parametrize("pipeline_config", pipeline_configs)
+@pytest.mark.precommit
+@pytest.mark.nightly
+def test_invalid_length_properties_raise_error(pipeline_config):
+    model_path = get_models_list()[0][1]
+    with pytest.raises(RuntimeError):
+        pipe = ov_genai.LLMPipeline(model_path, "NPU", pipeline_config)
+
+
+@pytest.mark.skipif(is_npu_available, reason="Not supposed to be run on NPU device")
+@pytest.mark.precommit
+@pytest.mark.nightly
+def test_batch_one_no_exception():
+    model_path = get_models_list()[0][1]
+    prompt = 'The Sun is yellow because'
+    static_pipe = ov_genai.LLMPipeline(model_path, "NPU")
+    # Check it doesn't throw any exception when batch of size 1 is provided
+    actual_out = static_pipe.generate([prompt], max_new_tokens=20)
+
+
+# TODO: For the further batch support
+@pytest.mark.skipif(is_npu_available, reason="Not supposed to be run on NPU device")
+@pytest.mark.precommit
+@pytest.mark.nightly
+def test_batch_raise_error():
+    model_path = get_models_list()[0][1]
+    prompt = 'The Sun is yellow because'
+    pipe = ov_genai.LLMPipeline(model_path, "NPU")
+    with pytest.raises(RuntimeError):
+        pipe.generate([prompt] * 3, max_new_tokens=100)
+
+
+# TODO: For the further sampling support
+generation_configs = [
+    dict(num_beam_groups=3),
+    dict(temperature=2),
+    dict(do_sample=True),
+    dict(top_p=0.8),
+    dict(top_k=10),
+]
+@pytest.mark.skipif(is_npu_available, reason="Not supposed to be run on NPU device")
+@pytest.mark.parametrize("generation_config", generation_configs)
+@pytest.mark.precommit
+@pytest.mark.nightly
+def test_unsupported_sampling_raise_error(generation_config):
+    model_path = get_models_list()[0][1]
+    prompt = 'The Sun is yellow because'
+    pipe = ov_genai.LLMPipeline(model_path, "NPU")
+    with pytest.raises(RuntimeError):
+        pipe.generate(prompt, **generation_config)
+
+
+@pytest.mark.skipif(is_npu_available, reason="Not supposed to be run on NPU device")
+@pytest.mark.precommit
+@pytest.mark.nightly
+def test_max_number_of_tokens(model_descr):
+    model_path = get_models_list()[0][1]
+    prompt = 'The Sun is yellow because'
+    num_tokens = 128
+
+    pipe = ov_genai.LLMPipeline(model_path, "NPU")
+    tokenizer = ov_genai.Tokenizer(model_path)
+    tokenized_input = tokenizer.encode(prompt)
+    # ignore_eos=True to ensure model will generate exactly num_tokens
+    encoded_results = pipe.generate(tokenized_input, max_new_tokens=num_tokens, ignore_eos=True)
+    assert len(encoded_results.tokens[0]) == num_tokens
+
+
+# FIXME: Known problem, output differs from stateful pipeline starting from 3rd prompt!
```

Comment:
Failed ticket already

Reply:
Filed

Reply:
Please, add a ticket number to the comment.

Reply:
Done, I'd omit real JIRA name though

Reply:
Sure. You can only leave the number